packages/@aws-cdk/aws-imagebuilder-alpha/lib/distribution-configuration.ts

```diff
@@ -0,0 +1,766 @@
+
+  /**
+   * The name of the distribution configuration
+   */
+  public readonly distributionConfigurationName: string;
+
+  private readonly amiDistributionsByRegion: { [region: string]: AmiDistribution } = {};
+  private readonly containerDistributionsByRegion: {
+    [region: string]: ContainerDistribution;
+  } = {};
+
+  public constructor(scope: Construct, id: string, props: DistributionConfigurationProps = {}) {
+    super(scope, id, {
+      physicalName:
+        props.distributionConfigurationName ??
+        cdk.Lazy.string({
+          produce: () =>
+            cdk.Names.uniqueResourceName(this, {
+              maxLength: 128,
+              separator: '-',
+              allowedSpecialCharacters: '-',
+            }).toLowerCase(), // Enforce lowercase for the auto-generated fallback
+        }),
+    });
+
+    Object.defineProperty(this, DISTRIBUTION_CONFIGURATION_SYMBOL, { value: true });
+
+    this.validateDistributionConfigurationName();
+
+    this.addAmiDistributions(...(props.amiDistributions ?? []));
+    this.addContainerDistributions(...(props.containerDistributions ?? []));
+
+    const distributionConfiguration = new CfnDistributionConfiguration(this, 'Resource', {
+      name: this.physicalName,
+      description: props.description,
+      distributions: cdk.Lazy.any({ produce: () => this.renderDistributions() }),
+      tags: props.tags,
+    });
+
+    this.distributionConfigurationName = this.getResourceNameAttribute(distributionConfiguration.attrName);
+    this.distributionConfigurationArn = this.getResourceArnAttribute(distributionConfiguration.attrArn, {
+      service: 'imagebuilder',
+      resource: 'distribution-configuration',
+      resourceName: this.physicalName,
+    });
+  }
+
+  /**
+   * Adds AMI distribution settings to the distribution configuration
+   *
+   * @param amiDistributions The list of AMI distribution settings to apply
+   */
+  public addAmiDistributions(...amiDistributions: AmiDistribution[]): void {
+    amiDistributions.forEach((amiDistribution) => {
+      const region = amiDistribution.region ?? cdk.Stack.of(this).region;
+      if (!cdk.Token.isUnresolved(region) && this.amiDistributionsByRegion[region]) {
+        throw new cdk.ValidationError(
+          `duplicate AMI distribution found for region "${region}"; only one AMI distribution per region is allowed`,
+          this,
+        );
+      }
+
+      this.amiDistributionsByRegion[region] = amiDistribution;
+    });
+  }
+
+  /**
+   * Adds container distribution settings to the distribution configuration
+   *
+   * @param containerDistributions The list of container distribution settings to apply
+   */
+  public addContainerDistributions(...containerDistributions: ContainerDistribution[]): void {
+    containerDistributions.forEach((containerDistribution) => {
+      const region = containerDistribution.region ?? cdk.Stack.of(this).region;
+      if (this.containerDistributionsByRegion[region]) {
```

Comment:
Token validation is missing here.
```suggestion
      if (!cdk.Token.isUnresolved(region) && this.containerDistributionsByRegion[region]) {